### PR TITLE
move scripts for install/update/remove hooks into rpm binary #40

### PR DESCRIPTION
**Pull request by brianhatchl**

### CODE REVIEW

**Comment by jasonsurratt** [comment]
Jenkins, I goofed. I restarted the postgres instance while you were testing. My apologies. test this please

**Comment by brianhatchl** [comment]
I think this really is a failure of the hoot-services-ui rpm post install script in setting up postgres.  The bash code was moved from an external file in the hoot repo into the rpm spec file for this PR.  I will dig in further.

**Comment by brianhatchl** [comment]
>==> default: /var/tmp/rpm-tmp.6DFueZ: line 193: warning: here-document at line 40 delimited by end-of-file (wanted `EOT')
==> default: /var/tmp/rpm-tmp.6DFueZ: line 194: syntax error: unexpected end of file
==> default: warning: %post(hootenanny-services-ui-0.2.23_577_g1395e08-1.el6.x86_64) scriptlet failed, exit status 2
==> default: Non-fatal POSTIN scriptlet failure in rpm package hootenanny-services-ui-0.2.23_577_g1395e08-1.el6.x86_64

**Comment by brianhatchl** [comment]
@jasonsurratt would you review and merge?

File: src/SPECS/hootenanny.spec.in [hunk 1]
```diff
@@ -169,44 +169,263 @@ This package contains the UI and web services.
...
+        sudo bash -c "cat >> $TOMCAT_ENV" <<EOT
+#--------------
+# Hoot Settings
+#--------------
+HOOT_HOME=/var/lib/hootenanny
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:$HOOT_HOME/lib:$HOOT_HOME/pretty-pipes/lib
+GDAL_DATA=/usr/local/share/gdal
+GDAL_LIB_DIR=/usr/local/lib
+HOOT_WORKING_NAME=hoot
+QT_HOME=/usr/local/qt-everywhere-opensource-src-4.8.6
```
Comment (jasonsurratt) [inline src/SPECS/hootenanny.spec.in]: Are these paths correct? We don't install in /usr/local w/ the RPMs.

Comment (brianhatchl) [inline src/SPECS/hootenanny.spec.in]: Those config settings came out of the https://github.com/ngageoint/hootenanny/blob/develop/scripts/InstallHoot-CentOS-RHEL.sh script, but maybe they are not correct when install gdal from rpm vs building from source.  And I guess the tests we run don't exercise any code that has hoot-services calling gdal.

I'll check the gdal spec file for correct paths.

Comment (jasonsurratt) [inline src/SPECS/hootenanny.spec.in]: Do they need to be explicitly set? HootEnv.sh should set them appropriately whenever a hoot command is run.

If you really need it you can get those values with:

export GDAL_DATA=`gdal-config --datadir`
export GDAL_LIB_DIR=`gdal-config --libs | sed -e "s/-L//g" | sed -e "s/ *-lgdal.*//g"`

Comment (jasonsurratt) [inline src/SPECS/hootenanny.spec.in]: Also, `LD_LIBRARY_PATH` should be unnecessary since we're installing in system lib directories.

Comment (brianhatchl) [inline src/SPECS/hootenanny.spec.in]: Hah!  Quite possible not.  Was just propagating what was in the CentOS install script.

Comment (jasonsurratt) [inline src/SPECS/hootenanny.spec.in]: Damn raisins.

File: src/SPECS/hootenanny.spec.in [hunk 2]
```diff
@@ -337,11 +318,11 @@ EOT
...
-        sudo iptables -A INPUT -p tcp -m state --state NEW -m tcp --dport 80 -j ACCEPT
-        sudo iptables -A INPUT -p tcp -m state --state NEW -m tcp --dport 8080 -j ACCEPT
-        sudo iptables -A INPUT -p tcp -m state --state NEW -m tcp --dport 8000 -j ACCEPT
-        sudo iptables -A PREROUTING -t nat -p tcp --dport 80 -j REDIRECT --to-ports 8080
-        sudo iptables -A OUTPUT -t nat -s 0/0 -d 127/8 -p tcp --dport 80 -j REDIRECT --to-ports 8080
+        sudo iptables -I -A INPUT -p tcp -m state --state NEW -m tcp --dport 80 -j ACCEPT
+        sudo iptables -I -A INPUT -p tcp -m state --state NEW -m tcp --dport 8080 -j ACCEPT
+        sudo iptables -I -A INPUT -p tcp -m state --state NEW -m tcp --dport 8000 -j ACCEPT
+        sudo iptables -I -A PREROUTING -t nat -p tcp --dport 80 -j REDIRECT --to-ports 8080
+        sudo iptables -I -A OUTPUT -t nat -s 0/0 -d 127/8 -p tcp --dport 80 -j REDIRECT --to-ports 8080
```
Comment (jasonsurratt) [inline src/SPECS/hootenanny.spec.in]: What does `-I -A` do? insert it at the beginning and append it?

Comment (brianhatchl) [inline src/SPECS/hootenanny.spec.in]: `-I -A` tells you that I didn't read the man page for iptables ;-)

I'll remove the append flag.

**Pull request merged**